kytos/core/connection.py

```diff
@@ -93,11 +93,13 @@ def close(self):
             try:
                 self.socket.shutdown(SHUT_RDWR)
                 self.socket.close()
+                self.socket = None
+                log.debug('Connection Closed: %s', self.id)
             except OSError as e:
                 if e.errno not in (ENOTCONN, EBADF):
                     raise e
-            self.socket = None
-            log.debug('Connection Closed: %s', self.id)
+            except AttributeError as e:
```

Comment:
Since we are catching the exception, we do not need to check for the attribute on line 91 anymore, and can unident the code. (see comment on the issue)